basket/base/tests/test_rq_utils.py

```diff
@@ -0,0 +1,239 @@
+import os
+from unittest.mock import patch
+
+from django.conf import settings
+from django.test import TestCase
+from django.test.utils import override_settings
+
+import pytest
+from rq.job import Job, JobStatus
+from rq.serializers import JSONSerializer
+
+from basket.base.rq import (
+    IGNORE_ERROR_MSGS,
+    get_queue,
+    get_redis_connection,
+    get_worker,
+    rq_exponential_backoff,
+    store_task_exception_handler,
+)
+from basket.base.tests.tasks import failing_job
+from basket.news.models import FailedTask
+
+
+class TestRQUtils(TestCase):
+    @override_settings(RQ_MAX_RETRIES=10)
+    def test_rq_exponential_backoff(self):
+        """
+        Test that the exponential backoff function returns the correct number
+        of retries, with a minimum of 60 seconds between retries.
+        """
+        with patch("basket.base.rq.random") as mock_random:
+            mock_random.randrange.side_effect = [2 * 2**n for n in range(settings.RQ_MAX_RETRIES)]
+            self.assertEqual(rq_exponential_backoff(), [60, 60, 60, 60, 60, 64, 128, 256, 512, 1024])
```

Comment:
That's only a total of around 40 minutes. Our current backoff can go on for days. Is this right, or can we extend it further?

Reply:
Ooh, good catch. In converting this I got mixed up b/c the celery code is working against minutes whereas I'm working against seconds. So I need to increase the factor here from `2 * 2**n` to `120 * 2**n` to be more inline with what we had with celery.

With that change this would be:
`[120, 240, 480, 960, 1920, 3840, 7680, 15360, 30720, 61440]`
without the jitter which looks better.

Thanks for checking my math.